### DISCUSSION TITLE
PnP Search Results - Accessibility issue: multiple <main> sections in a single page

### DISCUSSION THREAD

**Discussion by Luca3082** [microsoft-search/pnp-modern-search#3608] [Q&A]
Good morning,
We are using PnP Search Results within a SharePoint Online site to display search results.
Each page can contains multiple PnP Search Results webpart instances in order to show different search results.
As part of an accessibility assessment, the presence of multiple "main" tags in the same page was reported as critical, and by analyzing the problem we saw that each instance of the PnP Search Results webpart inserts its content into the page within a main tag.

![main section](https://github.com/microsoft-search/pnp-modern-search/assets/15248198/408bfd29-eb17-4c31-ace3-86996bafb2d9)

They told us that for accessibility reasons the main tag should appear only once on each page: is there a specific reason why PnP Search always adds this main section for every webpart occurrence?
Have you already faced this accessibility issue? And if so, do you have any suggestions in order to remove these sections?
Thank you

**Comment by wobba** [comment]
You can edit the template you use and change the markup to fit your scenario. And reason is likely for single page search setups.

**Reply by Luca3082** [reply]
Hi,
forgive me but it's not clear to me how to change the scenario.
As far as we know, our library of components to be inserted at the custom handlebars level draws the components after what is built in the Pnp Search Results webpart.
Correct me if I'm wrong but, analyzing the code of the webpart itself, we noticed that everything we can put in our templates is preceded by this main section.
![search results](https://github.com/microsoft-search/pnp-modern-search/assets/15248198/1f4fea3e-5a18-4c49-9269-6f602dd99809)

Is there a way that we obviously don't know about to remove this previous part from our template library?
Thank you

**Reply by wobba** [reply]
My bad... `<main>` is in the code as you point out and not the template. Can you suggest what we should use instead?

**Reply by Luca3082** [reply]
Well, I don't know if it's possible, but I think that the scenario in which you have more istances of the PnP Search Results webpart in a page is not uncommon, and the company that made that accessibility assessment told us that it's better to have zero "main" tags in a page rather than 3 or 4... So maybe the best thing is to remove the tag...

**Reply by wobba** [reply]
@Luca3082 you can do a PR for this if you want :)
